R/test-threads.R

```diff
@@ -330,90 +330,59 @@ adjust_estimation_options <- function(.mods, .cap_iterations){
     map(.mods, function(.mod){
 
       mod_path <- get_model_path(.mod)
-      mod_lines <- mod_path %>% readLines() %>%
-        suppressSpecificWarning("incomplete final line found")
-
+      ctl <- nmrec::read_ctl(mod_path)
 
       # Identify EST Blocks
-      est_idxs <- get_est_idx(mod_lines)
-
-      est_block <- map(seq_along(est_idxs), ~{
-        mod_lines[est_idxs[[.x]]]
-      })
-
-      est_idxs <- unlist(est_idxs)
+      ests <- nmrec::select_records(ctl, "est")
 
-      if(is.null(est_idxs)){
+      if(rlang::is_empty(ests)){
         warning(glue("No Estimation line found in {basename(mod_path)}, so there are no options to cap"))
         return(NULL)
       }
 
-      for(i in seq_along(est_block)){
-        est_block.i <- est_block[[i]]
-
-        max_match <- "MAX(EVAL(S)?)?=\\d+" ## need to vet this regex
-        max_detect <- str_detect(est_block.i, max_match)
-        niter_detect <- str_detect(est_block.i, "NITER=\\d+")
-        nburn_detect <- str_detect(est_block.i, "NBURN=\\d+")
-        if (any(max_detect)) {
-          est_block.i[max_detect] <- replace_est_opt(est_block.i[max_detect], max_match, .cap_iterations)
-        } else if(any(niter_detect)) {
-          est_block.i[niter_detect] <- replace_est_opt(est_block.i[niter_detect], "NITER=\\d+", .cap_iterations)
-          if (any(nburn_detect) && !any(str_detect(est_block.i, "NBURN=0\\b"))) {
-            est_block.i[nburn_detect] <- replace_est_opt(est_block.i[nburn_detect], "NBURN=\\d+", .cap_iterations)
-          }else if(!any(nburn_detect)){
+      for(i in seq_along(ests)){
+        est_block.i <- ests[[i]]
+        est_block.i$parse()
+
+        max_detect <- any(purrr::map_lgl(est_block.i$get_options(), ~ .x$name == "maxevals"))
+        niter_detect <- any(purrr::map_lgl(est_block.i$get_options(), ~ .x$name == "niter"))
+        nburn_detect <- any(purrr::map_lgl(est_block.i$get_options(), ~ .x$name == "nburn"))
+
+        if(max_detect){
+          nmrec::set_record_option(est_block.i, "maxevals", .cap_iterations)
+        }else if(niter_detect){
+          nmrec::set_record_option(est_block.i, "niter", .cap_iterations)
+          if(nburn_detect && !check_record_val(est_block.i, "nburn", 0)) {
+            # We don't want to overwrite NBURN if it was set to 0
+            nmrec::set_record_option(est_block.i, "nburn", .cap_iterations)
+          }else if(!nburn_detect){
             # NITER was specified, but NBURN wasn't (which means the default is being relied on). Force specification to cap.
-            message(glue("     Adding NBURN declaration to {est_block.i[niter_detect]} in {basename(mod_path)}"))
-            est_block.i[niter_detect] <- glue("{est_block.i[niter_detect]} NBURN={.cap_iterations}")
+            message(glue("Adding NBURN={.cap_iterations} declaration to {basename(mod_path)} to avoid using the default value"))
+            nmrec::set_record_option(est_block.i, "NBURN", .cap_iterations)
```

Comment:
All of the other spots that call `nmrec::set_record_option` are modifying an existing option, so they keep whatever name the caller used originally, which is good. This spot is adding a new option, so the spelling you give will go in.

Makes sense, and I like that you took care to go with upper case here because it's my impression that that's the predominant style for NONMEM options.

Reply:
Yeah I was conflicted with the case - thought about determining the case of other options, but that seemed a bit overkill